lib/Conversion/ATenToXTenPass.cpp

```diff
@@ -145,6 +144,53 @@ bool isLongestBranch(OpResult left, OpResult right) {
   return true;
 }
 
+// Tablegen adapter to check if attributes allow a conversion from `ReduceMean`
+// to `GlobalAveragePool`.
+bool isReduceMeanGlobalAveragePool2D(Value dims, Value keepdims) {
+
+  // Get defining operands
+  Torch::PrimListConstructOp dimsOp =
+      llvm::dyn_cast<Torch::PrimListConstructOp>(dims.getDefiningOp());
+  Torch::ConstantBoolOp keepdimsOp =
+      llvm::dyn_cast<Torch::ConstantBoolOp>(keepdims.getDefiningOp());
+
+  if (!dimsOp || !keepdimsOp)
+    return false;
+
+  SmallVector<APInt> axes;
+  for (Value operand : dimsOp.getOperands()) {
+    if (auto axesValue =
+            llvm::dyn_cast<Torch::ConstantIntOp>(operand.getDefiningOp()))
+      axes.push_back(axesValue.value());
+  }
+
+  // Check that arguments match expectations for global average pool
+  if (keepdimsOp.value() == 1 && axes.size() == 2 && axes[0] == 2 &&
+      axes[1] == 3)
+    return true;
+
+  return false;
+}
+
+// Tablegen adapter to check if attributes allow a conversion from `AveragePool`
+// to `GlobalAveragePool`.
+bool isAdaptiveAvgPoolGlobalAveragePool2D(Value outsizes) {
+  Torch::PrimListConstructOp outsizesOp =
+      llvm::dyn_cast<Torch::PrimListConstructOp>(outsizes.getDefiningOp());
+
+  SmallVector<APInt> dims;
+  for (Value operand : outsizesOp.getOperands()) {
+    if (auto axesValue =
+            llvm::dyn_cast<Torch::ConstantIntOp>(operand.getDefiningOp()))
```

Comment:
Is it okay to ignore operands that are not `Torch::ConstantIntOp`?

Reply:
Yes, `axes` values are always int constants in `torch-mlir`. See https://github.com/llvm/torch-mlir/blob/e23fbc89f27aa8ad190ce0903eb28a7fa0795c19/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L50 for example.

Reply:
When you are certain of the type I understand you should use `cast<>` (a checked cast) instead of `dyn_cast<>` (a checking cast) - https://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates.